esphome/components/wifi/__init__.py

```diff
@@ -157,9 +157,7 @@ def _validate(config):
         config[CONF_NETWORKS] = cv.ensure_list(WIFI_NETWORK_STA)(network)
 
     if (CONF_NETWORKS not in config) and (CONF_AP not in config):
-        raise cv.Invalid(
-            "Please specify at least an SSID or an Access Point " "to create."
-        )
+        config[CONF_NETWORKS] = cv.ensure_list(WIFI_NETWORK_STA)([])
```

Comment:
This should be enough:
```suggestion
        config[CONF_NETWORKS] = []
```